src/frontend/api.ts

```diff
@@ -256,12 +256,18 @@ export function Api({ baseUrl, timeout = DEFAULT_TIMEOUT }: ApiParam) {
       nodejs.channel.post("request-status");
       bugsnag.leaveBreadcrumb("Starting Mapeo Core");
       nodejs.start("loader.js");
-      const serverStartPromise = new Promise(resolve =>
-        // TODO: channel is supposed to extend RN's EventEmitter
-        // https://code.janeasystems.com/nodejs-mobile/react-native/bridge
-        //@ts-ignore
-        nodejs.channel.once("status", resolve)
-      ).then(async () => {
+      const serverStartPromise = new Promise<void>(resolve => {
+        // nodejs-mobile-react-native channel extends React Native's EventEmitter
+        //   - https://code.janeasystems.com/nodejs-mobile/react-native/bridge
+        const statusSubscription = nodejs.channel.addListener("status", () => {
+          // addListener returns an EventSubscription with the remove method
+          //   - https://github.com/facebook/react-native/blob/v0.66.3/Libraries/vendor/emitter/_EventEmitter.js#L57
+          //   - https://github.com/facebook/react-native/blob/v0.66.3/Libraries/vendor/emitter/_EventSubscription.js#L39
+          // @ts-expect-error
+          statusSubscription.remove();
```

Comment:
i _think_ the TS typings for nodejs-mobile-react-native need to be updated since the documentation seems to be correct. Maybe worth adding issue/PR for it on the repo as a follow-up

Reply:
yes can you do that?

Reply:
yeah i added an issue there for now: https://github.com/nodejs-mobile/nodejs-mobile-react-native/issues/3